src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -332,67 +345,66 @@ def _decode_content_disposition(self, content_disposition: bytes) -> dict[str, s
         parameters_dict = {p[0]: p[1].strip('"') for p in parameters}
         return parameters_dict
 
+    @staticmethod
     def _call_api(
-        self,
+        client: Optional[requests.Session],
         page: Tuple[io.BytesIO, int],
         request: requests.PreparedRequest,
         form_data: FormData,
         filename: str,
     ) -> requests.Response:
-        """
-        Calls the API with the provided parameters.
+        """Calls the API with the provided parameters.
+
+        The method is static to allow for parallel execution.
 
         Args:
-            page_content (Tuple[io.BytesIO, int]): A tuple containing the page content and
-            page number.
-            func (Callable): The function to call the API.
-            request (requests.PreparedRequest): The prepared request object.
-            form_data (FormData): The form data to include in the request.
-            filename (str): The name of the original file.
+            page: A tuple containing the page content and page number.
+            request: The prepared request object.
+            form_data: The form data to include in the request.
+            filename: The name of the original file.
 
         Returns:
             requests.Response: The response from the API.
 
         """
-        if self.client is None:
+        if client is None:
             raise RuntimeError("HTTP client not accessible!")
         page_content, page_number = page
 
-        new_request = self._create_request(
+        new_request = SplitPdfHook._create_request(
             request, form_data, page_content, filename, page_number
         )
-        prepared_request = self.client.prepare_request(new_request)
+        prepared_request = client.prepare_request(new_request)
 
         try:
-            return self.client.send(prepared_request)
+            return client.send(prepared_request)
         except Exception:
             logger.error("Failed to send request for page %d", page_number)
             return requests.Response()
 
+    @staticmethod
     def _create_request(
-        self,
         request: requests.PreparedRequest,
         form_data: FormData,
         page_content: io.BytesIO,
         filename: str,
         page_number: int,
     ) -> requests.Request:
-        """
-        Creates a request object for a part of a splitted PDF file.
+        """Creates a request object for a part of a splitted PDF file.
 
         Args:
-            request (requests.PreparedRequest): The original request object.
```

Comment:
we have type hints already, so this is unnecessary IMO

Reply:
Yeah, I was using some vscode extension that adds these automatically when generating doc template, that's why type hints are here (i believe that was me that added documentation here). I have nothing against removing it.

Reply:
alright, I'm removing them